### ISSUE TITLE
Efficient and clear migrate command that allows us to migrate from lower version to a higher version

### ISSUE THREAD

**Issue by aprilmintacpineda** [aws-amplify/amplify-cli#4849]
>**Note**: If your feature-request is regarding the AWS Amplify Console service, please log it in the 
[official AWS Amplify Console forum](https://forums.aws.amazon.com/forum.jspa?forumID=314&start=0)

**Is your feature request related to a problem? Please describe.**

Migrating is unclear and inefficient, I just tried to migrate from v4.22 to v 4.24 and when I tried to do `amplify update function` to update one of my functions, that was previously working but no longer working, looks like the permission to s3 storage was removed, so I tried to update it to re-add that permission, I noticed something odd on `function-parameters.json`, it looks like this:

```json
{
  "permissions": {
    "auth": {
      "myappauthentication": [
        "create",
        "read",
        "update",
        "delete"
      ]
    },
    "api": {
      "myappgraphqlapi": [
        "create",
        "read",
        "update",
        "delete"
      ]
    }
  },
  "mutableParametersState": {
    "permissions": {
      "auth": {
        "myappauthentication": [
          "create",
          "read",
          "update",
          "delete"
        ]
      },
      "api": {
        "myappgraphqlapi": [
          "create",
          "read",
          "update",
          "delete"
        ]
      },
      "storage": {
        "myapps3bucket": [
          "create",
          "read",
          "update",
          "delete"
        ]
      }
    }
  },
  "lambdaLayers": []
}
```

notice that there's `permissions` key which was there before I did the update, it doesn't have storage, hence the reason my function exited with a 403 error, after the update I notice it now has `mutableParametersState` key, which seems to be redundant with `permissions` except `permissions` still doesn't have the `storage` permission, which indicates that it was not updated during the update process and maybe the code now uses a different key and therefore the `permissions` key is no longer needed but was not removed, now I'm afraid that my other functions might not work IF that `permissions` key is no longer being used, now what I have to do is thoroughly test every functions that I have which will be inconvenient, imagine migrating from v4 to v5 or something like that, without this your app is locked to a specific version and when you need something that a higher version has then you're pretty much trapped, I'm not 100% sure about it, it's just that there's no clear and effective way of migrating lower version to a higher version.

**Describe the solution you'd like**

A clear instruction of how to migrate from lower to higher version.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

**Comment by UnleashedMind** [comment]
@aprilmintacpineda 
Thanks for your feedbacks. Our team will look into this. 
Were you able to get the function working?

**Comment by jhockett** [comment]
Hi @aprilmintacpineda, you are correct the `permissions` key is no longer used. Somehow this scenario was missed in our migration tests. The `permissions` entry can be removed in `function-parameters.json` if you want, though it shouldn't cause any issue if it's left as is.

The best immediate mitigation would be to run `amplify update function` for all of your Lambda functions and reapply the permissions you want.

**Comment by edwardfoyle** [comment]
Hi @aprilmintacpineda this is actually a bug. The mutableParametersState field was never intended to be written to the file and as such no migration would have been necessary. We're working on a fix, but in the meantime you can mitigate the issue with @jhockett's suggestion above

**Comment by aprilmintacpineda** [comment]
Thank you all for your suggestions. Yes, I was able to migrate the functions and make them work.

**Comment by edwardfoyle** [comment]
Reopening because there's still a related bug that we need to fix

**Comment by edwardfoyle** [comment]
@aprilmintacpineda The fix for this has been released in CLI v 4.26.0
